src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexInfo.cs

```diff
@@ -35,31 +32,21 @@ namespace System.Text.RegularExpressions.Symbolic
                 }
             }
 
-            if (startsWithLineAnchor || containsLineAnchor || startsWithBoundaryAnchor || containsSomeAnchor)
+            if (containsSomeAnchor || startsWithLineAnchor || startsWithSomeAnchor)
             {
                 i |= ContainsSomeAnchorMask;
 
-                if (startsWithLineAnchor || containsLineAnchor)
+                if (startsWithLineAnchor)
                 {
-                    i |= ContainsLineAnchorMask;
-
-                    if (startsWithLineAnchor)
-                    {
-                        i |= StartsWithLineAnchorMask;
-                    }
+                    i |= StartsWithLineAnchorMask;
                 }
 
-                if (startsWithBoundaryAnchor)
+                if (startsWithLineAnchor || startsWithSomeAnchor)
                 {
-                    i |= StartsWithBoundaryAnchorMask;
+                    i |= StartsWithSomeAnchorMask;
                 }
```

Comment:
Nit:
To retain the style of the rest of the checks, this could be:
```C#
if (startsWithLineAnchor || startsWithSomeAnchor)
{
    i |= StartsWithSomeAnchorMask;

    if (startsWithLineAnchor)
    {
        i |= StartsWithLineAnchorMask
    }
}
```